Source/WebCore/html/CheckboxInputType.cpp

```diff
@@ -104,23 +104,25 @@ void CheckboxInputType::handleKeyupEvent(KeyboardEvent& event)
 
 void CheckboxInputType::handleMouseDownEvent(MouseEvent& event)
 {
-    ASSERT(element());
+    RefPtr element = this->element();
+    ASSERT(element);
 
-    if (!event.isTrusted() || !isSwitch() || element()->isDisabledFormControl() || !element()->renderer())
+    if (!event.isTrusted() || !isSwitch() || element->isDisabledFormControl() || !element->renderer())
```

Comment:
Maybe we should split into 2 if checks so we don't Ref the element unnecessarily in the early return case:
```
if (!event.isTrusted() || !isSwitch())
    return;

RefPtr element = this->element();
ASSERT(element);
if (element->isDisabledFormControl() || !element->renderer()))
    return;
```

Reply:
Good idea, done.